capstone/src/main/webapp/script.js

```diff
@@ -12,44 +12,219 @@
+  'pt': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT']
+};
+
+/** Collects the user's input and produces Perspective analysis */
+async function handleInput() {
+  // Get the submitted text and language
+  const textElement = document.getElementById('textForAnalysis');
+  if (!textElement) {
     return;
   }
-  outputElement.innerHTML = '';
+  const langElement = document.getElementById('languageForAnalysis');
+  if (!langElement) {
+    return;
+  }
+
+  // Get the selected attributes
+  const attributes = document.getElementById("available-attributes").getElementsByTagName('input');
+  const requestedAttributes = []
+  for (let attribute of attributes) {
+    if (attribute.checked) {
+      requestedAttributes.push(attribute.value);	
+    }	
+  }
+
+  // Make Perspective call for the entire submission and load data
+  const toxicityData = await callPerspective(textElement.value, langElement.value, requestedAttributes);
+  loadChartsApi(toxicityData);
+
+  // Draw the separating line for the output
+  const separator = document.getElementById('separator-container');
+  separator.innerHTML = '';
+  separator.appendChild(document.createElement('hr'));
+
+  // Get detailed analysis if requested
+  document.getElementById('analysis-container').innerHTML = '';
+  const radios = document.getElementsByName('analysisRadios');
+  var delimiter = null;
+  for (i = 0; i < radios.length; i++) {
+    if (radios[i].checked) {
+      delimiter = radios[i].value;
+      break;
+    }
+  }
+  if (delimiter != "") {
+    getAnalysis(textElement.value, langElement.value, requestedAttributes, delimiter); 
+  }
+}
 
-  if (toxicityData.attributeScores) {
-    for (let key in toxicityData.attributeScores) {
-      if (toxicityData.attributeScores[key].summaryScore && 
-          toxicityData.attributeScores[key].summaryScore.value) {
-        attributeElement = createAnyElement('p', key + ": " + toxicityData.attributeScores[key].summaryScore.value);
-        outputElement.appendChild(attributeElement);
+/** Prints detailed analysis of text by word or sentence */
+async function getAnalysis(text, lang, requestedAttributes, delimiter) {
+  // Set up the detailed analysis section
+  const analysisContainer = document.getElementById('analysis-container')
+  analysisContainer.appendChild(createAnyElement('b', 'Detailed Anaylsis'));
+  const result = createAnyElement('p', '');
+  result.className = 'detailed-analysis';
+  
+  // Generate the results for every substring of the input text
+  const substrings = text.split(delimiter);
+  for (i = 0; i < substrings.length; i++) {
+    if (substrings[i] != '') {
+      // Get the Perspective scores for the substring & sort them descending
+      const response = await callPerspective(substrings[i], lang, requestedAttributes);
+      const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
```

Comment:
There's the possibility that we might need to add error checking here in case Perspective fails.

Reply:
Added error checking